aten/src/ATen/cuda/CUDAFuture.cpp

```diff
@@ -27,9 +27,24 @@ namespace {
 std::vector<std::reference_wrapper<const at::DataPtr>> extractDataPtrs(
     const at::IValue& value) {
   at::IValue::HashAliasedIValues sub_values;
-  // Prefer getSubValues() over visit() as the latter is a silent no-op for
-  // some unsupported types, whereas the former at least fails loudly.
-  value.getSubValues(sub_values);
+  try {
+    // Prefer getSubValues() over visit() as the latter is a silent no-op for
+    // some unsupported types, whereas the former at least fails loudly.
+    value.getSubValues(sub_values);
+  } catch (const c10::TypeError& err) {
+    // We might not be able to extract subvalues from Python objects (e.g., if
+    // they are custom classes) but we can still extract tensors.
```

Comment:
Does this mean `ConcretePyObjectHodler::extractTensors` is on a best-effort basis and can't extract tensors that might be contained in nested classes?

Reply:
It is "best effort" in the sense that while it widely enlarges the set of objects that are now supported, there are still objects that we won't be able to handle. These are however just the unpicklable objects, which hopefully are quite uncommon. Note that if you have an unpicklable object you're already unable to use it, for example, as an argument or a return value of a RPC call, hence I think it's reasonable to also say you cannot use it as a return value of a callback.

> can't extract tensors that might be contained in nested classes?

This should be supported, since pickle will just enter those nested classes and inspect them too. If you have a specific example you're worried about I'd be happy to add it to the tests in #56518.

Reply:
Ah ok, sounds good, I just misunderstood the comment.